drivers/interrupt_controller/Kconfig.nrf5

```diff
@@ -0,0 +1,32 @@
+# Nordic Semiconductor interrupt controller configuration
+
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# Workaround for not being able to have commas in macro arguments
+DT_COMPAT_NORDIC_NRF_EGU := nordic,nrf-egu
+
+config INTC_NRF_EGU
+	bool "EGU interrupt driver"
+	depends on $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_EGU))
+	depends on NRFX_EGU
+	help
+	  Enable support for EGU interrupt driver for nRF MCU series.
+
+choice INTC_SWI
+	prompt "Software interrupt implementation"
+	help
+	  Enables software interrupt utility. The software interrupt can be used
+	  to de-escalate interrupt processing from zero-latency interrupts to a
+	  priority level at which it is safe to use kernel services.
+
+DT_CHOSEN_SWI_NRF_EGU := nordic,swi-egu
```

Comment:
Upstream is not an appropriate place to have a chosen node specific to a vendor like this.

Are you sure you don't mean to be using a compatible here instead?